m3/reporter_test.go

```diff
@@ -222,6 +222,88 @@ func TestReporterFinalFlush(t *testing.T) {
 	require.Equal(t, 1, len(server.Service.getBatches()[0].GetMetrics()))
 }
 
+func TestReporterHistogram(t *testing.T) {
+	var wg sync.WaitGroup
+	server := newFakeM3Server(t, &wg, true, Compact)
+	go server.Serve()
+	defer server.Close()
+
+	r, err := NewReporter(Options{
+		HostPorts:          []string{server.Addr},
+		Service:            "test-service",
+		CommonTags:         defaultCommonTags,
+		MaxQueueSize:       queueSize,
+		MaxPacketSizeBytes: maxPacketSize,
+		Interval:           0,
+	})
+	require.NoError(t, err)
+
+	wg.Add(1)
+
+	h := r.AllocateHistogram("my-histogram", map[string]string{
+		"foo": "bar",
+	}, tally.Durations([]time.Duration{
+		0 * time.Millisecond,
+		25 * time.Millisecond,
+		50 * time.Millisecond,
+		75 * time.Millisecond,
+		100 * time.Millisecond,
+	}))
+	b := h.DurationBucket(0*time.Millisecond, 25*time.Millisecond)
+	b.ReportSamples(7)
+	b = h.DurationBucket(50*time.Millisecond, 75*time.Millisecond)
+	b.ReportSamples(3)
+	r.Close()
+
+	wg.Wait()
+
+	require.Equal(t, 1, len(server.Service.getBatches()))
+	require.NotNil(t, server.Service.getBatches()[0])
+	require.Equal(t, 2, len(server.Service.getBatches()[0].GetMetrics()))
+
+	// Verify first bucket
+	counter := server.Service.getBatches()[0].GetMetrics()[0]
+	require.Equal(t, "my-histogram", counter.GetName())
+	require.True(t, counter.IsSetTags())
+	require.Equal(t, 3, len(counter.GetTags()))
+	for tag := range counter.GetTags() {
+		require.Equal(t, map[string]string{
+			"foo":      "bar",
+			"bucketid": "1",
+			"bucket":   "0-25ms",
+		}[tag.GetTagName()], tag.GetTagValue())
+	}
+	require.True(t, counter.IsSetMetricValue())
+	val := counter.GetMetricValue()
+	require.True(t, val.IsSetCount())
+	require.False(t, val.IsSetGauge())
+	require.False(t, val.IsSetTimer())
+	count := val.GetCount()
+	require.True(t, count.IsSetI64Value())
+	require.EqualValues(t, int64(7), count.GetI64Value())
```

Comment:
Why use `EqualValues` instead of `Equal` here? You're already explicitly casting the constant to `int64`.